app/components/osc-webhook-triggers/osc-webhook-triggers.component.js

```diff
@@ -20,9 +21,59 @@
   });
 
 
-  function OscWebhookTriggers($scope, $uibModal, $filter, APIService) {
+  function OscWebhookTriggers($filter, $scope, $timeout, $uibModal, APIService) {
     var ctrl = this;
 
+    // Check if the webhook trigger contains only deprecated secret format.
+    ctrl.isDeprecated = function(trigger) {
+      var triggerSecretData = $filter('getWebhookSecretData')(trigger);
+      return _.has(triggerSecretData, 'secret') && !_.has(triggerSecretData, 'secretReference.name');
+    };
+
+    ctrl.addEmptyWebhookTrigger = function() {
+      ctrl.webhookTriggers.push({
+        lastTriggerType: "",
+        data: {
+          type: ""
+        }
+      });
+      var newTriggerIndex = ctrl.webhookTriggers.length - 1;
+      $timeout(function() {
+        $scope.$broadcast('triggerTypeFocus' + newTriggerIndex);
+      });
+    };
+
+    // Check if new or modified webhook trigger is a duplicate. If so, show a warning under the appropriate trigger
+    var checkDuplicates = function(trigger) {
+      var matchingTriggers = _.filter(ctrl.webhookTriggers, function(wehbookTrigger) {
```

Comment:
Check duplicates needs to avoid checking values when some of the data is not set. See the changes I had in the PR to your branch here: https://github.com/jhadvig/origin-web-console/pull/1/commits/0424279e9228b060b6e49469f7cd61cac9afe2e8#diff-9d51eb270d91c5c6831894971b883c5cR48